src/AggregateFunctions/AggregateFunctionGroupArraySorted.h

```diff
@@ -0,0 +1,390 @@
+#pragma once
+
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/Operators.h>
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <AggregateFunctions/AggregateFunctionGroupArray.h>
+#include <Functions/array/arraySort.h>
+
+#include <Common/Exception.h>
+#include <Common/ArenaAllocator.h>
+#include <Common/assert_cast.h>
+#include <Columns/ColumnConst.h>
+#include <DataTypes/IDataType.h>
+#include <base/sort.h>
+#include <Columns/IColumn.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+
+#include <algorithm>
+#include <type_traits>
+#include <utility>
+
+#define AGGREGATE_FUNCTION_GROUP_ARRAY_MAX_ARRAY_SIZE 0xFFFFFF
+
+
+namespace DB
+{
+struct Settings;
+
+namespace ErrorCodes
+{
+    extern const int TOO_LARGE_ARRAY_SIZE;
+    extern const int INCORRECT_DATA;
+}
+
+template <typename T, typename Trait>
+class GroupArraySortedNumericImpl final
+    : public IAggregateFunctionDataHelper<GroupArrayNumericData<T, false>, GroupArraySortedNumericImpl<T, Trait>>
```

Comment:
Let's remove this inheritance.

GroupArrayNumericData is a very simple class that contains PODArray. (Also it contains total_values which is non needed here, only makes the state bigger).

Reply:
Still relevant